### PR TITLE
py-typer: add version 0.15.1 and "standard" optional dependencies

### PR DESCRIPTION
**Pull request by lgarrison**
https://github.com/fastapi/typer/tree/0.15.1

### CODE REVIEW

**Comment by lgarrison** [comment]
@spackbot re-run pipeline

**Comment by lgarrison** [comment]
@spackbot re-run pipeline

File: var/spack/repos/builtin/packages/py-typer/package.py [hunk 1]
```diff
@@ -12,12 +12,25 @@ class PyTyper(PythonPackage):
     homepage = "https://github.com/tiangolo/typer"
     pypi = "typer/typer-0.9.0.tar.gz"
 
-    license("MIT")
+    license("MIT", checked_by="lgarrison")
 
+    variant("standard", default=True, description="Optional dependencies installed by default")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-typer/package.py]: Please move the variant directive to below the version directives.

File: var/spack/repos/builtin/packages/py-typer/package.py [hunk 2]
```diff
@@ -12,12 +12,25 @@ class PyTyper(PythonPackage):
...
-    depends_on("python@3.6:", type=("build", "run"))
-    depends_on("py-flit-core@2.0:2", type="build")
-    depends_on("py-click@7.1.1:8", type=("build", "run"))
+    with when("@0.15.1:"):
+        depends_on("python@3.7:", type=("build", "run"))
+        depends_on("py-pdm-backend", type="build")
+        depends_on("py-click@8:", type=("build", "run"))
+
+        depends_on("py-shellingham@1.3.0:", type=("build", "run"), when="+standard")
+        depends_on("py-rich@10.11.0:", type=("build", "run"), when="+standard")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-typer/package.py]: Slightly confused. According to the `pyproject.toml` file, these don't appear to be optional:

```
dependencies = [
    "click >= 8.0.0",
    "typing-extensions >= 3.7.4.3",
    "shellingham >=1.3.0",
    "rich >=10.11.0",
]
```

File: var/spack/repos/builtin/packages/py-typer/package.py [hunk 3]
```diff
@@ -12,12 +12,25 @@ class PyTyper(PythonPackage):
...
 
-    depends_on("python@3.6:", type=("build", "run"))
-    depends_on("py-flit-core@2.0:2", type="build")
-    depends_on("py-click@7.1.1:8", type=("build", "run"))
+    with when("@0.15.1:"):
+        depends_on("python@3.7:", type=("build", "run"))
+        depends_on("py-pdm-backend", type="build")
+        depends_on("py-click@8:", type=("build", "run"))
+
+        depends_on("py-shellingham@1.3.0:", type=("build", "run"), when="+standard")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-typer/package.py]:
```suggestion
        depends_on("py-shellingham@1.3:", type=("build", "run"), when="+standard")
```

File: var/spack/repos/builtin/packages/py-typer/package.py (hunk 2 above)
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-typer/package.py]:
```suggestion
        depends_on("py-rich@10.11:", type=("build", "run"), when="+standard")
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the sha256 and dependencies. 

Don't need trailing `.0` in dependency ranges. Unclear, based on `pyproject.toml` if the new variant is needed but, if it is, then that directive should be listed *after* all version directives.

**Comment by lgarrison** [comment]
Regarding the variant, it looks like `standard` is an optional dependency that only exists in the source, not the sdist! I think the custom `pdm_build.py` script is inserting those dependencies into the `pyproject.toml` at build time. Since we're using the sdist, I agree that this shouldn't be a variant. Thanks for catching that.

And thank you for the advice about `.0`; I wasn't sure if best practice was to match the exact version numbers from `pyproject.toml` as closely as possible, or whether to simplify/normalize version numbers. I'll do the latter going forward.

**Review by tldahlgren — APPROVED** [review]
Thanks.

**Pull request merged**